docs/porting.md

```diff
@@ -77,6 +79,10 @@ Called when the device received SET_CONFIG request, you can leave this empty if
 ##### dcd_remote_wakeup
 Called to remote wake up host when suspended (e.g hid keyboard)
 
+##### dcd_connect / dcd_disconnect
+
+Connect or disconnect the data-line pull-up resistor. Define as a weak function if the MCU has an internal pull-up, otherwise do not define.
```

Comment:
if PU is available, it will be the strong one, BSP/application has no purpose to use external PU, even if it does, it is not a concern of TinyUSB stack.

Reply:
I was trying to find a use case, but I remembered somewhat wrong. The STM32H742 has an errata stating its internal pull-up is too small and will wear out after 10 years, so someone could conceivably want to use the external one instead.

Regardless, will modify this PR based on feedback.

Reply:
wow, that errata for 10 years wear-out is insane though